### PR TITLE
build: switch to clang (PROOF-620)

### PR DESCRIPTION
**Pull request by rnburn**
# Rationale for this change

Nvcc doesn't have good support for coroutines and crashes on valid code. We've put in bug reports, but we're migrating to clang in the meantime.

# What changes are included in this PR?

* switch cuda compiler from nvcc to clang
* update bazel to use the official cuda rules: https://github.com/bazel-contrib/rules_cuda

# Are these changes tested?

Uses existing tests.

### CODE REVIEW

File: sxt/base/macro/cuda_warning.h [hunk 1]
```diff
@@ -16,7 +16,8 @@
  */
 #pragma once
 
-#ifdef __CUDACC__
+// #ifdef __CUDACC__
```
Comment (joe-stifler) [inline sxt/base/macro/cuda_warning.h:19]: Nit: do you intend to remove this comment?

File: sxt/memory/resource/BUILD [hunk 2]
```diff
@@ -14,8 +14,8 @@ sxt_cc_component(
     with_test = False,
     deps = [
         "//sxt/base/type:raw_stream",
-        "@local_config_cuda//cuda:cuda_headers",
-        "@local_config_cuda//cuda:cudart_static",
+        # "@local_config_cuda//cuda:cuda_headers",
```
Comment (joe-stifler) [inline sxt/memory/resource/BUILD:17]: Nit: and what about those comments?

File: sxt/base/device/BUILD [hunk 3]
```diff
@@ -162,10 +162,10 @@ sxt_cc_component(
         ":property",
         "//sxt/base/test:unit_test",
     ],
-    deps = [
-        "@local_config_cuda//cuda:cuda_headers",
-        "@local_config_cuda//cuda:cudart_static",
-    ],
+    # deps = [
```
Comment (joe-stifler) [inline sxt/base/device/BUILD:165]: Nit: and those comments?

File: .bazelrc [hunk 4]
```diff
@@ -1,11 +1,31 @@
...
+build --flag_alias=cuda_copts=@rules_cuda//cuda:copts
+build --flag_alias=cuda_host_copts=@rules_cuda//cuda:host_copts
+build --flag_alias=cuda_runtime=@rules_cuda//cuda:runtime
+
+build --@rules_cuda//cuda:archs=compute_70:compute_70,sm_70
+build --@rules_cuda//cuda:runtime=@local_cuda//:cuda_runtime_static
+
+build --enable_cuda=True
+
+# Use --config=clang to build with clang instead of gcc and nvcc.
```
Comment (joe-stifler) [inline .bazelrc:21]: Do you intend to update CI in a follow-up PR to use this `--config=clang` flag?

**Review by joe-stifler — DISMISSED** [review]
Thanks, Ryan!

File: .bazelrc (hunk 4 above)
Comment (rnburn) [reply .bazelrc:21]: It's set in the .bazelrc

File: sxt/base/macro/cuda_warning.h (hunk 1 above)
Comment (rnburn) [reply sxt/base/macro/cuda_warning.h]: Fixed .

File: sxt/memory/resource/BUILD (hunk 2 above)
Comment (rnburn) [reply sxt/memory/resource/BUILD]: Removed. Thanks.

File: sxt/base/device/BUILD (hunk 3 above)
Comment (rnburn) [reply sxt/base/device/BUILD]: Also removed. Thanks.

**Review by joe-stifler — APPROVED** [review]

**Pull request merged**